common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala

```diff
@@ -140,6 +148,20 @@ object WhiskPackage
     val bindingFieldName = "binding"
     override val collectionName = "packages"
 
+    /**
+     * an utility function that traverses all potential binding references and returns the resolved package
+     * @param pkg the package for which the resolution to be applied
+     * @return the same package if there is no binding, or the fully resolved package
+     */
+    def resolveBinding(entityStore: EntityStore, pkg: DocId)(
```

Comment:
It depends on how it is used. For one action, there will not be such a sequence. For a sequence of actions, you'll have such a layer.